### PR TITLE
Bindings: Add extensions to the yarp::sig::Sound class

### PR DESCRIPTION
**Pull request by Bontempogianpaolo1**
Improvement to easily export yarp sounds to python

### CODE REVIEW

**Comment by drdanz** [comment]
@Bontempogianpaolo1 Thanks for the patch! Can you please add a small description of your changes to the release notes as described by the bot here https://github.com/robotology/yarp/pull/2426#issuecomment-736753846 ?

**Review by traversaro — COMMENTED** [review]
See inline comment

**Comment by randaz81** [comment]
I'd suggest to add also the other two functions for **interleaved vectors**. It's super easy: they just have the two for loops swapped.

File: bindings/yarp.i [hunk 1]
```diff
@@ -627,6 +628,37 @@ typedef yarp::os::BufferedPort<Vector> BufferedPortVector;
...
+        vec.reserve(m_samples*channels);
+        for (size_t c = 0; c < channels; c++)
+        {
+            for (size_t t = 0; t < m_samples; t++)
+            {
+                vec.push_back(self->get(t, c));
+            }
+        }
+        return vec;
+ }
```
[the remark this answers is not in the document]
Comment (drdanz) [reply bindings/yarp.i:647]: The number of bracket is correct, the indentation is a bit confusing... @Bontempogianpaolo1 Do you mind fixing it?

[the remark this answers is not in the document]
Comment (Bontempogianpaolo1) [reply bindings/yarp.i:647]: Sure no problem! How does it work? should i resolve conflicts and make another pull request? Is the branch name correct? I have just noticed the big difference from other branches. I would take this opportunity to add the functions described [here](https://github.com/robotology/yarp/pull/2426#issuecomment-737098145)

[the remark this answers is not in the document]
Comment (drdanz) [reply bindings/yarp.i:647]: Just add a new commit to the branch, and the PR will be updated

**Review by drdanz — APPROVED** [review]
LGTM now

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**